src/rgw/rgw_op.cc

```diff
@@ -3876,6 +3876,16 @@ void RGWPutObj::execute(optional_yield y)
   CompressorRef plugin;
   boost::optional<RGWPutObj_Compress> compressor;
 
+  obj_ctx.alloc_hint_flags =
+    s->cct->_conf.get_val<bool>("rgw_enable_alloc_hint");
+  auto compression_hint = s->cct->_conf.get_val<std::string>("rgw_compression_hint");
```

Comment:
how can a global config option know whether specific data is likely to be compressible?  this seems like a model that wouldn't be very generally applicable